src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -834,14 +842,36 @@ namespace winrt::TerminalApp::implementation
         // Since the previous focus location might be discarded in the background,
         // e.g., the command palette will be dismissed by the menu,
         // and then closing the fly-out will move the focus to wrong location.
-        newTabFlyout.Opening([this](auto&&, auto&&) {
-            _FocusCurrentTab(true);
+        newTabFlyout.Opening([weakThis{ get_weak() }](auto&&, auto&&) {
+            if (auto page{ weakThis.get() })
```

Comment:
this is only necessary if there's a chance the hosting page goes away before the dropdown button goes away. generally for UI-speed things (like the user clicking on something) all this weak reference stuff is just noise honestly.